rover_auxiliary/src/ipcamera_api.py

```diff
@@ -0,0 +1,322 @@
+        'LensShadeCorrection': ('LSC', 11),
+        'LensDistortionCorrection': ('LDC', 15),
+        'LensDistortionCorrectionLevel': ('LDCLevel', 19),
+        'AntiFog': ('AFG', 10),
+        'AntiFogLevel': ('AFGLevel', 20),
+        'SceneSelect': ('scene_Select', 2),
+        'ExposureMode': ('AEMode_Select', 2),
+        'ShutterSpeed': ('ShutterSpeedText', 2),
+        'ManualACG': ('AEGainText', 2),
+        'IRenable': ('IRenable', 0),
+        'IRmode': ('IRmode', 0),
+        'WhiteBalanceModeSelect': ('WBMode_Select', 3)
+    }
+
+    RESOLUTION_MAP = {
+        '480p': (640, 480),
+        'w480p': (720, 480),
+        '576p': (720, 576),
+        '720p': (1280, 720),
+        '960p': (1280, 960),
+        '1080p': (1920, 1080),
+        '1536p': (2048, 1536)
+    }
+
+    SHUTTER_MAP = {
+        "1/8000": 0,
+        "1/6000": 1,
+        "1/4000": 2,
+        "1/2000": 3,
+        "1/1000": 4,
+        "1/500": 5,
+        "1/250": 6,
+        "1/200": 7,
+        "1/150": 8,
+        "1/100": 9,
+        "1/50": 10,
+        "1/25": 11,
+        "1/20": 12,
+        "1/15": 13,
+        "1/10": 14,
+        "1/8": 15,
+        "1/5": 16,
+        "1/3": 17,
+        "1/2": 18,
+        "1": 19
+    }
+
+    AEGAIN_MAP = {
+        "1X": 0,
+        "2X": 1,
+        "4X": 2,
+        "8X": 3,
+        "16X": 4,
+        "32X": 5,
+        "64X": 6
+    }
+
+    def __init__(self, ip: str, port: int, username: str, password: str, timeout: int = 5):
+        self.ip = ip
+        self.port = port
+        self.username = username
+        self.password = password
+
+        # Configure logging
+        for handler in logging.root.handlers[:]:
+            logging.root.removeHandler(handler)
+        logging.basicConfig(level=logging.INFO, format='%(levelname)s - %(message)s')
+        self.logger = logging.getLogger()
+
+        def timeout_handler(signum, frame):
+            raise TimeoutException("Camera initialization timed out")
+
+        signal.signal(signal.SIGALRM, timeout_handler)
+        signal.alarm(timeout)
+
+        try:
+            self.camera = ONVIFCamera(self.ip, self.port, self.username, self.password)
+
+            self.media_service = self.camera.create_media_service()
+            profiles = self.media_service.GetProfiles()
+            if not profiles:
+                raise Exception("No media profiles found on the camera")
+            self.profile = profiles[0]
+
+            self.imaging_service = self.camera.create_imaging_service()
+            self.source_token = self.profile.VideoSourceConfiguration.SourceToken
+
+        except TimeoutException:
+            self.logger.error("Camera initialization timed out")
+        except Exception as e:
+            self.logger.error(f"Failed to initialize the camera: {str(e).split(':')[-1].strip()}")
+        finally:
+            signal.alarm(0)  
+
+    def set_onvif_param(self, param: str, value: str, extra: tuple=None):
```

Comment:
Can this be declared private? Since there's no checks on params str or values, it would make sure a user of the lib doesn't call this in the wrong way.